pkg/apis/resource/validation/validation.go

```diff
@@ -279,9 +278,14 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		}
 	}
 
-	// Updates to a populated status.Allocation are not allowed
+	// Updates to a populated status.Allocation are not allowed.
+	// Unmodified fields don't need to be validated again and,
+	// in this particular case, must not be validated again because
+	// validation for new results is tighter than it was before.
 	if oldStatus.Allocation != nil && status.Allocation != nil {
 		allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(status.Allocation, oldStatus.Allocation, fldPath.Child("allocation"))...)
+	} else if status.Allocation != nil {
+		allErrs = append(allErrs, validateAllocationResult(status.Allocation, fldPath.Child("allocation"), requestNames)...)
```

Comment:
Is this related to this change? I can see why its needed, but it seems unrelated to providing AdminAccess field as part of the DeviceRequestAllocationResult. Does it at least deserve its own commit?

Reply:
We only needed it because of the special logic that is required for fields that might be invalid on an update, so it belongs in the same commit? Otherwise we probably would validate again.